connmgr/connmanager.go

```diff
@@ -399,9 +422,12 @@ func (cm *ConnManager) Stop() {
 // New returns a new connection manager.
 // Use Start to start connecting to the network.
 func New(cfg *Config) (*ConnManager, error) {
-	if cfg.Dial == nil {
+	if (cfg.Dial == nil) && (cfg.DialAddr == nil) {
```

Comment:
No need for parenthesis.  Please remove them so it's more consistent with the rest of the code.